Gems/Terrain/Code/Source/TerrainSystem/TerrainSystem.cpp

```diff
@@ -551,13 +713,16 @@ void TerrainSystem::ProcessHeightsFromList(
         return;
     }
 
+    AZStd::vector<bool> terrainExists(inPositions.size());
+    AZStd::vector<float> heights(inPositions.size());
```

Comment:
Just as a thought, instead of heights as a vector<float>, it _might_ be faster to make this a vector<AZ::Vector3> for the positions. Instead of reserving the sizes here and passing into GetHeightsSynchronous as a span<>, you'd just pass the empty vector<>s in and let that method do the reserving.  GetHeightsSynchronous at the end would then overwrite into itself as it goes along, and then resize down to the proper size at the end.

If nothing else, it would use even less memory... :)